h5netcdf/core.py

```diff
@@ -316,6 +351,27 @@ def __len__(self):
     @property
     def datatype(self):
         """Return numpy dtype or user defined type."""
+        if self._root._h5py.__name__ == "h5py":
```

Comment:
This is the current procedure to extract the committed type as h5netcdf type. For h5pyd and transient types special handling is needed.